File: ThefieldofdreamsClient/src/app/models/wallet.ts
export type TransactionType = 'Credit' | 'Debit' | 'Reward' | 'Purchase' | 'Refund';
export type TransactionStatus = 'Pending' | 'Completed' | 'Failed' | 'Cancelled';

export interface Wallet {
  id: string;
  userId: string;
  balance: number;
  points: number;
  stripeCustomerId?: string;
}

export interface Transaction {
  id: string;
  walletId: string;
  amount: number;
  type: TransactionType;
  description?: string;
  referenceId?: string;
  status: TransactionStatus;
  createdAt: string;
}

export interface CreateTransactionRequest {
  walletId: string;
  amount: number;
  type: TransactionType;
  description?: string;
  referenceId?: string;
}

File: Thefieldofdreams.Domain/Entities/Wallet.cs
using System.Transactions;

namespace Thefieldofdreams.Domain.Entities
{
    

    public class Wallet : BaseEntity
    {
        public Guid UserId { get; set; }
        public decimal Balance { get; set; } = 0;
        public int Points { get; set; } = 0;
        public string? StripeCustomerId { get; set; }

        public virtual User User { get; set; }
        public virtual ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
    }

    public class Transaction : BaseEntity
    {
        public Guid WalletId { get; set; }
        public decimal Amount { get; set; }
        public TransactionType Type { get; set; }
        public string? Description { get; set; }
        public string? ReferenceId { get; set; }
        public TransactionStatus Status { get; set; } = TransactionStatus.Pending;

        public virtual Wallet Wallet { get; set; }
    }

    public enum TransactionType
    {
        Credit,
        Debit,
        Reward,
        Purchase,
        Refund
    }

    public enum TransactionStatus
    {
        Pending,
        Completed,
        Failed,
        Cancelled
    }
}